prelude/slang-hlsl-prelude.h

```diff
@@ -1,4 +1,13 @@
 #ifdef SLANG_HLSL_ENABLE_NVAPI
 #include "nvHLSLExtns.h"
 #endif
-#pragma warning(disable: 3557)
+
+#ifdef __DXC_VERSION_MAJOR
+    // warning: '&' within '|' [-Wbitwise-op-parentheses]
+    #pragma dxc diagnostic ignored "-Wbitwise-op-parentheses"
+    // warning: operator '>>' has lower precedence than '+'; '+' will be evaluated first [-Wshift-op-parentheses]
+    #pragma dxc diagnostic ignored "-Wshift-op-parentheses"
+#else
+    // warning X3557: loop doesn't seem to do anything, forcing loop to unroll
```

Comment:
not 100% related to this issue, but worth bringing up:

I wonder if we should add a similar warning in the front end:  "The loop only executes for a limited number of iterations or doesn't seem to do anything so consider removing it or forcing it to unroll."

and we also add a second warning: "unrolling may reduce performance in large shaders due to register pressure" if a loop is large and running for many iterations.

Reply:
Instead of silencing the warning, we should consider changing our emit logic to always emit a parenthesis if the operator is &, | or << >>.

Reply:
Interesting. I thought we were intentionally not emitting the parentheses. I will see if I can emit them.

Reply:
> not 100% related to this pull request, but worth bringing up since we are silencing warnings:
> 
> I wonder if we should add a similar warning in the front end: "The loop only executes for a limited number of iterations or doesn't seem to do anything so consider removing it or forcing it to unroll."
> 
> and we also add a second warning: "unrolling may reduce performance in large shaders due to register pressure" if a loop is large and running for many iterations.

I believe we have the warning already; or similar